src/Features/Core/Portable/CodeRefactorings/InvertIf/AbstractInvertIfCodeRefactoringProvider.Analyzer.cs

```diff
@@ -0,0 +1,298 @@
+                        }
+                        else
+                        {
+                            return InvertIfStyle.MoveIfBodyToElseClause;
+                        }
+                    }
+                }
+            }
+
+            private bool IsEmptySubsequence(TIfStatementSyntax ifStatement)
+            {
+                foreach (var range in GetSubsequentStatementRange(ifStatement))
+                {
+                    if (!IsEmptyStatementRange(range))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+
+            private bool IsEmptyIfBody(TIfStatementSyntax ifStatement)
+            {
+                return IsEmptyStatementRange(GetIfBodyStatementRange(ifStatement));
+            }
+
+            private bool SingleIfBodyStatement(TIfStatementSyntax ifStatement)
+            {
+                var range = GetIfBodyStatementRange(ifStatement);
+                return range.first == range.last;
+            }
+
+            private bool SingleSubsequenceStatement(TIfStatementSyntax ifStatement)
+            {
+                using (var e = GetSubsequentStatementRange(ifStatement).GetEnumerator())
+                {
+                    return e.MoveNext() && e.Current.first == e.Current.last && !e.MoveNext();
+                }
+            }
+
+            private async Task<Document> InvertIfAsync(
+                Document document,
+                TIfStatementSyntax ifStatement,
+                InvertIfStyle invertIfStyle,
+                SyntaxNode subsequenceSingleExitPointOpt,
+                CancellationToken cancellationToken)
+            {
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                return document.WithSyntaxRoot(
+                    // this returns the root because VB requires changing context around if statement
+                    GetRootWithInvertIfStatement(
+                        document, semanticModel, ifStatement, invertIfStyle, subsequenceSingleExitPointOpt, cancellationToken));
+            }
+
+            private void AnalyzeSubsequenceControlFlow(
+                SemanticModel semanticModel,
+                TIfStatementSyntax ifStatement,
+                out bool subsequenceEndPontIsReachable,
+                out SyntaxNode subsequenceSingleExitPointOpt)
+            {
+                subsequenceEndPontIsReachable = true;
+                subsequenceSingleExitPointOpt = null;
+
+                foreach (var range in GetSubsequentStatementRange(ifStatement))
+                {
+                    AnalyzeControlFlow(semanticModel, range, out subsequenceEndPontIsReachable, out subsequenceSingleExitPointOpt);
+                    if (!subsequenceEndPontIsReachable)
+                    {
+                        return;
+                    }
```

Comment:
fwiww, i find this pattern of out-vars and checking along the way confusing.  I somewhat feel like just having tuples would be nicer.  but i don't feel strongly about it.

Reply:
(wontfix)